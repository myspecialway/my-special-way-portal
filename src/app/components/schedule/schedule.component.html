<table class="mat-table schedule-table" role="grid">
  <thead>
    <tr class="mat-header-row" role="row">
      <th class="mat-header-cell cdk-column-position mat-column-position" role="columnheader" colspan="2">שעה</th>
      <th class="mat-header-cell cdk-column-position mat-column-position" role="columnheader" *ngFor="let day of daysLabels">{{day}}</th>
    </tr>
  </thead>
  <tbody>
    <tr class="mat-row" role="row" *ngFor="let row of schedule; index as hourIndex">
      <td class="mat-cell cdk-column-position mat-column-position" role="gridcell">{{hourIndex + 1}}</td>
      <td class="mat-cell cdk-column-position mat-column-position" role="gridcell">{{hoursLabels[hourIndex]}}</td>
      <td 
        class="mat-cell cdk-column-position mat-column-position"
        *ngFor="let timeslot of row; index as dayIndex" 
        role="gridcell" 
        (click)="timeSlotClicked.emit({hourIndex: hourIndex, dayIndex: dayIndex})"
      >
        <div *ngIf="timeslot.lesson" class="ts-full">
          <span class="ts-lesson">{{timeslot.lesson.title}}</span>
          <span class="ts-location" *ngIf="timeslot.location">{{timeslot.location.name}}</span>
        </div>
        <div *ngIf="!timeslot.lesson" class="ts-empty">
          <i class="material-icons" >
            add
          </i>
        </div>
      </td>
    </tr>
  </tbody>
</table>
