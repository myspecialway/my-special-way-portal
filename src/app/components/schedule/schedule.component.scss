$hover: rgba(106, 0, 207, 0.2);

.schedule-table {
    width: 100%;
    direction: rtl;
    font-family: "Rubik", "Helvetica", "Arial", sans-serif;
    color: #222;
    th.mat-header-cell {
        background-color: #fafafa;
        font-size: 16px;
        font-weight: bold;
        color: #222;
        border: 0;
    }
    td.mat-cell, td.mat-footer-cell, th.mat-header-cell {
        text-align: right;
        padding: 20px 14px;
    }
    td.mat-cell, td.mat-footer-cell {
        cursor: pointer;
        border: 1px solid #e0e0e0;
        font-size: 18px;
        &:hover {
            background-color: $hover;
        }
    }

    td.mat-cell:first-child, td:nth-child(2) {
        text-align: center;
        direction: initial;
        font-weight: 400;
    }
    
    .ts-lesson, .ts-location {
        font-family: Rubik;
        color: #222;
        display: block;
    }
    .ts-lesson {
        font-size: 18px;
        font-weight: 400;
    }
    .ts-location {
        font-size: 14px;
    }
    .ts-empty {
        text-align: center;
        color: #6a00cf;
        opacity: .5;
        &:hover {
            opacity: 1;
        }
    }
}
