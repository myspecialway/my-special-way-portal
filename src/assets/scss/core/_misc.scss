body {
  background-color: #fafafa;
  color: $black-color;
  font-weight: 300;
}

legend {
  border-bottom: 0;
}

.serif-font {
  font-family: $font-family-serif;
}

// Prevent highlight on mobile
* {
  -webkit-tap-highlight-color: rgba(255, 255, 255, 0);
  -webkit-tap-highlight-color: transparent;
  &:focus {
    outline: 0;
  }
}

a {
  color: $link-color;
  &:hover,
  &:focus {
    color: darken($link-color, 5%);
    text-decoration: none;
  }

  &.text-info {
    &:hover,
    &:focus {
      color: darken($brand-info, 5%);
    }
  }

  & .material-icons {
    vertical-align: middle;
  }
}

.form-check,
label {
  font-size: 14px;
  line-height: 1.42857;
  color: $checkboxes-text-color;
  font-weight: 400;
}

/*           Animations              */
.animation-transition-general {
  @include transition-all($general-transition-time, $transition-linear);
}

.animation-transition-slow {
  @include transition-all($slow-transition-time, $transition-linear);
}

.animation-transition-fast {
  @include transition-all($fast-transition-time, $transition-ease);
}

.caret,
.sidebar a {
  @include transition-all($fast-transition-time, $transition-ease-in);
}
#map {
  position: relative;
  z-index: 2;
  width: 100%;
  height: calc(100vh - 70px);
  margin-top: 70px;
}

@font-face {
  font-family: 'pass';
  font-style: normal;
  font-weight: 400;
  src: url(data:application/font-woff;charset=utf-8;base64,d09GRgABAAAAAATsAA8AAAAAB2QAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAABGRlRNAAABWAAAABwAAAAcg9+z70dERUYAAAF0AAAAHAAAAB4AJwANT1MvMgAAAZAAAAA/AAAAYH7AkBhjbWFwAAAB0AAAAFkAAAFqZowMx2N2dCAAAAIsAAAABAAAAAQAIgKIZ2FzcAAAAjAAAAAIAAAACAAAABBnbHlmAAACOAAAALkAAAE0MwNYJ2hlYWQAAAL0AAAAMAAAADYPA2KgaGhlYQAAAyQAAAAeAAAAJAU+ATJobXR4AAADRAAAABwAAAAcCPoA6mxvY2EAAANgAAAAEAAAABAA5gFMbWF4cAAAA3AAAAAaAAAAIAAKAE9uYW1lAAADjAAAARYAAAIgB4hZ03Bvc3QAAASkAAAAPgAAAE5Ojr8ld2ViZgAABOQAAAAGAAAABuK7WtIAAAABAAAAANXulPUAAAAA1viLwQAAAADW+JM4eNpjYGRgYOABYjEgZmJgBEI2IGYB8xgAA+AANXjaY2BifMg4gYGVgYVBAwOeYEAFjMgcp8yiFAYHBl7VP8wx/94wpDDHMIoo2DP8B8kx2TLHACkFBkYA8/IL3QB42mNgYGBmgGAZBkYGEEgB8hjBfBYGDyDNx8DBwMTABmTxMigoKKmeV/3z/z9YJTKf8f/X/4/vP7pldosLag4SYATqhgkyMgEJJnQFECcMOGChndEAfOwRuAAAAAAiAogAAQAB//8AD3jaY2BiUGJgYDRiWsXAzMDOoLeRkUHfZhM7C8Nbo41srHdsNjEzAZkMG5lBwqwg4U3sbIx/bDYxgsSNBRUF1Y0FlZUYBd6dOcO06m+YElMa0DiGJIZUxjuM9xjkGRhU2djZlJXU1UDQ1MTcDASNjcTFQFBUBGjYEkkVMJCU4gcCKRTeHCk+fn4+KSllsJiUJEhMUgrMUQbZk8bgz/iA8SRR9qzAY087FjEYD2QPDDAzMFgyAwC39TCRAAAAeNpjYGRgYADid/fqneL5bb4yyLMwgMC1H90HIfRkCxDN+IBpFZDiYGAC8QBbSwuceNpjYGRgYI7594aBgcmOAQgYHzAwMqACdgBbWQN0AAABdgAiAAAAAAAAAAABFAAAAj4AYgI+AGYB9AAAAAAAKgAqACoAKgBeAJIAmnjaY2BkYGBgZ1BgYGIAAUYGBNADEQAFQQBaAAB42o2PwUrDQBCGvzVV9GAQDx485exBY1CU3PQgVgIFI9prlVqDwcZNC/oSPoKP4HNUfQLfxYN/NytCe5GwO9/88+/MBAh5I8C0VoAtnYYNa8oaXpAn9RxIP/XcIqLreZENnjwvyfPieVVdXj2H7DHxPJH/2/M7sVn3/MGyOfb8SWjOGv4K2DRdctpkmtqhos+D6ISh4kiUUXDj1Fr3Bc/Oc0vPqec6A8aUyu1cdTaPZvyXyqz6Fm5axC7bxHOv/r/dnbSRXCk7+mpVrOqVtFqdp3NKxaHUgeod9cm40rtrzfrt2OyQa8fppCO9tk7d1x0rpiQcuDuRkjjtkHt16ctbuf/radZY52/PnEcphXpZOcofiEZNcQAAeNpjYGIAg///GBgZsAF2BgZGJkZmBmaGdkYWRla29JzKggxD9tK8TAMDAxc2D0MLU2NjENfI1M0ZACUXCrsAAAABWtLiugAA)
    format('woff');
}

input[type='password'] {
  font-family: 'pass', 'Roboto', Helvetica, Arial, sans-serif;
  font-size: 18px;
  &::-webkit-input-placeholder {
    transform: scale(0.77);
    transform-origin: 0 50%;
  }
  &::-moz-placeholder {
    font-size: 14px;
    opacity: 1;
  }
  &:-ms-input-placeholder {
    font-size: 14px;
    font-family: 'Roboto', Helvetica, Arial, sans-serif;
  }
}
