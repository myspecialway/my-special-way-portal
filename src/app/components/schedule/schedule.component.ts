import { Component, Input, Output, EventEmitter } from '@angular/core';
import { TimeSlot } from '../../models/timeslot.model';

export interface TimeSlotIndexes {
  hourIndex: number;
  dayIndex: number;
}

@Component({
  selector: 'app-schedule',
  templateUrl: './schedule.component.html',
  styleUrls: ['./schedule.component.scss'],
})

export class ScheduleComponent {

  @Input()
  schedule: TimeSlot[][];
  @Input()
  daysLabels: string[];
  @Input()
  hoursLabels: string[];
  @Output()
  timeSlotClicked: EventEmitter<TimeSlotIndexes> = new EventEmitter();
}
