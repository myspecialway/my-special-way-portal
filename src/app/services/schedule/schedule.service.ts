import { Injectable } from '@angular/core';

@Injectable()
export class ScheduleService {
  levels = ['א', 'ב', 'ג', 'ד', 'ה', 'ו'];
  daysLabels = ['ראשון', 'שני', 'שלישי', 'רביעי', 'חמישי', 'שישי'];
  hoursLabels = [
    '07:30 - 08:00',
    '08:00 - 08:50',
    '08:50 - 09:35',
    '09:35 - 10:00',
    '10:00 - 10:30',
    '10:30 - 11:15',
    '11:15 - 12:00',
    '12:00 - 12:45',
    '12:45 - 13:15',
    '13:15 - 13:45',
    '13:45 - 14:30',
    '14:30 - 15:15',
    '15:15 - 16:00',
    '16:00 - 16:30',
    '16:30 - 16:45',
  ];
}
