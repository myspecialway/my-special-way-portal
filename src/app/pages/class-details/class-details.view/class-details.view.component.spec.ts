import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ClassDetailsViewComponent } from './class-details.view.component';
import { NO_ERRORS_SCHEMA } from '@angular/compiler/src/core';
import { ScheduleService } from '../../../services/schedule/schedule.service';
import { TranslateLoader, TranslateModule } from '@ngx-translate/core';
import { TranslateCustomLoader } from '../../../../mocks/translate.stub';

const beforeEachAsync = async () => {
  TestBed.configureTestingModule({
    imports: [
      TranslateModule.forRoot({
        loader: {
          provide: TranslateLoader,
          useClass: TranslateCustomLoader,
        },
      }),
    ],
    declarations: [ClassDetailsViewComponent],
    providers: [ScheduleService],
    schemas: [NO_ERRORS_SCHEMA],
  }).compileComponents();
};

describe('ClassDetailsComponent with class info', () => {
  let fixture: ComponentFixture<ClassDetailsViewComponent>;
  let component: ClassDetailsViewComponent;
  beforeEach(beforeEachAsync);

  beforeEach(() => {
    fixture = TestBed.createComponent(ClassDetailsViewComponent);
    component = fixture.componentInstance;
    component.name = 'פטל';
    component.grade = 'a';
    component.shouldShowClassInfo = true;
    fixture.detectChanges();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});

describe('ClassDetailsComponent without class info', () => {
  let fixture: ComponentFixture<ClassDetailsViewComponent>;
  let component: ClassDetailsViewComponent;
  beforeEach(beforeEachAsync);

  beforeEach(() => {
    fixture = TestBed.createComponent(ClassDetailsViewComponent);
    component = fixture.componentInstance;
    component.name = 'פטל';
    component.grade = 'a';
    component.shouldShowClassInfo = false;
    fixture.detectChanges();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
