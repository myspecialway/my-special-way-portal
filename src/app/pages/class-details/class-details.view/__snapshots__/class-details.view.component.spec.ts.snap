// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassDetailsComponent with class info should match snapshot 1`] = `
<app-class-details-view
  _grade={[Function String]}
  _name={[Function String]}
  detailChanged={[Function EventEmitter]}
  gradesKeys={[Function Array]}
  scheduleService={[Function ScheduleService]}
  shouldShowClassInfo={[Function Boolean]}
  timeslotClicked={[Function EventEmitter]}
>
  <div
    class="main-content"
    data-test-id="class-details-page"
  >
    <div
      class="container-fluid "
    >
      <div
        class="row"
      >
        <div
          class="col-lg-12 col-md-12 col-sm-12 class-details-container"
        >
          
          <div>
            <a
              class="back"
            >
              <i
                class="material-icons"
              >
                chevron_right
              </i>
              כל הכיתות
            </a>
            <div
              class="class-details"
            >
              <div
                class="class-detail-field"
              >
                <mat-form-field>
                  <input
                    data-test-id="class-name-input"
                    matinput=""
                    placeholder="שם הכיתה"
                    required=""
                    type="text"
                  />
                  <i
                    class="material-icons"
                    matsuffix=""
                  >
                    mode_edit
                  </i>
                </mat-form-field>
              </div>
              <div
                class="class-detail-field"
              >
                <mat-form-field
                  class="class-detail-field"
                >
                  <mat-select
                    data-test-id="grade-select"
                    placeholder="שכבה"
                  >
                    
                    <mat-option
                      class="grade-select-option"
                      selected="key === grade"
                    >
                       GRADES.A 
                    </mat-option>
                    <mat-option
                      class="grade-select-option"
                      selected="key === grade"
                    >
                       GRADES.B 
                    </mat-option>
                    <mat-option
                      class="grade-select-option"
                      selected="key === grade"
                    >
                       GRADES.C 
                    </mat-option>
                    <mat-option
                      class="grade-select-option"
                      selected="key === grade"
                    >
                       GRADES.D 
                    </mat-option>
                    <mat-option
                      class="grade-select-option"
                      selected="key === grade"
                    >
                       GRADES.E 
                    </mat-option>
                    <mat-option
                      class="grade-select-option"
                      selected="key === grade"
                    >
                       GRADES.F 
                    </mat-option>
                  </mat-select>
                </mat-form-field>
              </div>
            </div>
          </div>
          <h2>
            מערכת שעות כיתת פטל
          </h2>
          <app-schedule />
        </div>
      </div>
    </div>
  </div>
</app-class-details-view>
`;

exports[`ClassDetailsComponent without class info should match snapshot 1`] = `
<app-class-details-view
  _grade={[Function String]}
  _name={[Function String]}
  detailChanged={[Function EventEmitter]}
  gradesKeys={[Function Array]}
  scheduleService={[Function ScheduleService]}
  shouldShowClassInfo="false"
  timeslotClicked={[Function EventEmitter]}
>
  <div
    class="main-content"
    data-test-id="class-details-page"
  >
    <div
      class="container-fluid "
    >
      <div
        class="row"
      >
        <div
          class="col-lg-12 col-md-12 col-sm-12 class-details-container"
        >
          
          <h2>
            מערכת שעות כיתת פטל
          </h2>
          <app-schedule />
        </div>
      </div>
    </div>
  </div>
</app-class-details-view>
`;
